controllers/volumereplicationgroup_controller.go

```diff
@@ -897,6 +907,22 @@ func (v *VRGInstance) processAsPrimary() (ctrl.Result, error) {
 	return ctrl.Result{}, nil
 }
 
+func (v *VRGInstance) handleVRGMode(state ramendrv1alpha1.ReplicationState) bool {
+	asyncNeedRequeue := false
+
+	if v.instance.Spec.Async.Mode == ramendrv1alpha1.AsyncModeEnabled {
+		if state == ramendrv1alpha1.Primary {
+			asyncNeedRequeue = v.reconcileVRsAsPrimary()
+		}
+
+		if state == ramendrv1alpha1.Secondary {
+			asyncNeedRequeue = v.reconcileVRsAsSecondary()
+		}
+	}
+
+	return asyncNeedRequeue
```

Comment:
handleVRGMode function could have additional return their respective if blocks. In case replicationstate  is primary it can directly return from there instead of going to the next heck.

Reply:
@raghavendrabhat This comment is probably valid. Need to address in a new PR?

Reply:
I think this is automatically handled in a subsequent PR where there is addition of sync mode. With that, in situations where both sync and async modes are there, we need to capture whether requeue is needed or not, separately for sync and async and then return true if either of them needs requeue. i.e. with sync mode, it becomes like this

===============================================================================
if v.instance.Spec.Async.Mode == ramendrv1alpha1.AsyncModeEnabled {
		if state == ramendrv1alpha1.Primary {
			asyncNeedRequeue = v.reconcileVRsAsPrimary()
		}

		if state == ramendrv1alpha1.Secondary {
			asyncNeedRequeue = v.reconcileVRsAsSecondary()
		}
	}

	if v.instance.Spec.Sync.Mode == ramendrv1alpha1.SyncModeEnabled {
		// mark all PVCs as protected.
		v.markAllPVCsProtected()

		syncNeedRequeue = false
	}

	return asyncNeedRequeue || syncNeedRequeue
===============================================================================

Above changes are there in PR #370. So as per that, we should not return from if blocks until we have handled both syncMode and asyncMode.